File: Bypyhton/删除有序数组中的重复项2.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        if len(nums) <= 2:
            return len(nums)

        n = 2
        i = 2
        while i < len(nums):
            if nums[n - 2] != nums[i]:
                nums[n] = nums[i]
                n += 1
            i += 1
        
        return n

File: ByCpp/删除有序数组中的重复项2.cpp
#include <vector>

using namespace std;

int removeDuplicates(vector<int>& nums) {
    if (nums.size() < 3) return nums.size();
    int n = 2;
    for (int i = 2; i < nums.size(); i++) {
        if (nums[n - 2] != nums[i]) {
            nums[n++] = nums[i];
        }
    }
    return n;
}
